src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java

```diff
@@ -78,4 +82,43 @@ public void doPostCleanup(BibEntry entry) {
     public Parser getParser() {
         return new BibtexParser(importFormatPreferences, new DummyFileUpdateMonitor());
     }
+
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) {
+        List<BibEntry> results = new ArrayList<>();
+        String doi = entry.getField(StandardField.DOI).orElse(null);
+
+        if (doi == null) {
```

Comment:
Optional already provides methods like isPresent ...to check if the optional has a value. And the purpose of Optional is to avod nulls
https://www.baeldung.com/java-optional